### PR TITLE
Controller dialog: Update controllers after controller addon installation

### PR DESCRIPTION
**Pull request by garbear**
## Description

This PR is a continuation of the UX fix in https://github.com/xbmc/xbmc/pull/15754.

The first commit adds the missing add-on events that caused the list to not refresh when an add-on is installed.

However, this led to the following problem:

Screen Shot 2019-03-16 at 5 18 30 PM

The SNES controller was selected, but installing a 3DO controller updated the controller list indices and caused the wrong one to be highlighted. To fix this, I made the controller dialog focus the newly installed controller.

## How Has This Been Tested?
Installed a controller and observed it being focused.

Clicked "Get all", and controllers were updated as they were installed. Unfortunately, the image is blocked by the modal dialog. I would like to make it a toast, but then how would the user cancel? Closing the dialog?

Screen Shot 2019-03-17 at 9 40 06 AM

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/games/controllers/windows/GUIControllerList.cpp [hunk 1]
```diff
@@ -75,8 +75,17 @@ void CGUIControllerList::Deinitialize(void)
   m_controllerButton = nullptr;
 }
 
-bool CGUIControllerList::Refresh(void)
+bool CGUIControllerList::Refresh(const std::string &controllerId)
```
Comment (Rechi) [inline xbmc/games/controllers/windows/GUIControllerList.cpp]: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types
```suggestion
bool CGUIControllerList::Refresh(const std::string& controllerId)
```

File: xbmc/games/controllers/windows/GUIControllerList.h [hunk 2]
```diff
@@ -35,7 +35,7 @@ namespace GAME
     // implementation of IControllerList
     virtual bool Initialize(void) override;
     virtual void Deinitialize(void) override;
-    virtual bool Refresh(void) override;
+    virtual bool Refresh(const std::string &controllerId) override;
```
Comment (Rechi) [inline xbmc/games/controllers/windows/GUIControllerList.h]:
```suggestion
    virtual bool Refresh(const std::string& controllerId) override;
```

File: xbmc/games/controllers/windows/IConfigurationWindow.h [hunk 3]
```diff
@@ -64,9 +64,10 @@ namespace GAME
 
     /*!
      * \brief Refresh the contents of the list
+     * \param controllerId The controller to focus, or empty to leave focus unchanged
      * \return True if the list was changed
      */
-    virtual bool Refresh(void) = 0;
+    virtual bool Refresh(const std::string &controllerId) = 0;
```
Comment (Rechi) [inline xbmc/games/controllers/windows/IConfigurationWindow.h]:
```suggestion
    virtual bool Refresh(const std::string& controllerId) = 0;
```

**Comment by garbear** [comment]
@enen92 can you test?

**Comment by enen92** [comment]
Sorry for the delay @garbear. Tested it and working as expected 👍 
One small glitch though: suppose you install all the available controllers. The "Get More" button is still enabled which leads the user to think there are more controllers to install. However if you click the button you'll only find out that "You already have installed all the controllers". Supposedly, the `UpdateButtons()` call in `GUIControllerWindow.cpp` is there to prevent this situation I guess.

**Comment by garbear** [comment]
> Which leads me to think that the subscription of addon install events I previously added in #15754 would still make some sense.

Agreed. I haven't had time to include that fix in this PR, so I think it's best if we merge as-is and follow up with a fix to the problem that #15754 tries to solved. Can you rebase on master after this merge so we can re-evaluate your patch?

jenkins build this please

**Comment by ksooo** [comment]
@garbear we are just in the middle of branching Leia. In case you want this fix in v18 you need to backport it to Leia branch.

File: xbmc/games/controllers/windows/GUIControllerList.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/games/controllers/windows/GUIControllerList.cpp]: https://github.com/xbmc/xbmc/pull/15763/commits

I assumed I squashed the commits a long time ago, but it looks like I was wrong.

**Comment by garbear** [comment]
@ksooo I saw the branching discussion in slack too late. Do you recommend backporting?

**Pull request merged**